server/src/main/java/org/jboss/as/server/operations/ServerSuspendHandler.java

```diff
@@ -99,24 +106,36 @@ public void suspendStarted() {
                     @Override
                     public void complete() {
                         suspendController.removeListener(this);
-                        context.completeStep(new RollbackHandler(suspendController));
+                        latch.countDown();
                     }
 
                     @Override
                     public void cancelled() {
                         suspendController.removeListener(this);
-                        context.setRollbackOnly();
-                        context.completeStep(new RollbackHandler(suspendController));
+                        cancelled.set(true);
+                        latch.countDown();
                     }
 
                     @Override
                     public void timeout() {
                         suspendController.removeListener(this);
-                        context.completeStep(new RollbackHandler(suspendController));
+                        latch.countDown();
                     }
                 };
                 suspendController.addListener(operationListener);
                 suspendController.suspend(timeout > 0 ?  timeout * 1000 : timeout);
+                if(timeout != 0) {
+                    try {
+                        latch.await();
+                    } catch (InterruptedException e) {
+                        Thread.currentThread().interrupt();
+                        throw new OperationFailedException(e);
```

Comment:
You don't need/want to throw an OFE. When this method returns, the OperationContext will detect the interrupt status on the thread and fall into its cancellation handling. The OFE will just introduce noise.